controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -234,6 +245,24 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// ClusterRegistryConfig represents registry config used with the cluster.
 	// +optional
 	ClusterRegistryConfig *RegistryConfig `json:"clusterRegistryConfig,omitempty"`
+
+	// AutoNode set the AutoNode mode and AutoNode role ARN.
+	// +optional
+	AutoNode *AutoNode `json:"autoNode,omitempty"`
+}
+
+// AutoNode set the AutoNode mode and AutoNode role ARN.
+type AutoNode struct {
+	// AutoNode mode allowed values are enabled & disabled
+	// +kubebuilder:validation:Enum=enabled;disabled
+	// +kubebuilder:default=disabled
+	// +optional
+	Mode AutoNodeMode `json:"mode,omitempty"`
```

Comment:
Let's be specific here on the description, what `Mode` means and what the default value is.
I know there is the kubebuilder directive but that's not visible in the godoc hints.